io.github.timcharper.EmojiClu.yml

```diff
@@ -0,0 +1,46 @@
+app-id: io.github.timcharper.EmojiClu
+runtime: org.gnome.Platform
+runtime-version: "47"
+sdk: org.gnome.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.rust-stable
+command: emojiclu
+tags:
+  - "$(cargo metadata --no-deps --format-version 1 | jq -r '.packages[0].version')"
+finish-args:
+  - --socket=fallback-x11
+  - --share=ipc
+  - --socket=wayland
+  - --socket=pulseaudio
+  - --device=dri
+modules:
+  - name: emojiclu
+    buildsystem: simple
+    build-options:
+      append-path: /usr/lib/sdk/rust-stable/bin
+      build-args:
+        - --share=network
+      env:
+        CARGO_HOME: /run/build/emojiclu/cargo
+        CARGO_NET_OFFLINE: 'true'
+    sources:
+      - type: git
+        url: https://github.com/timcharper/EmojiClu.git
+        tag: v1.0.1
+        commit: 3d458a60ebd401084e6ce75c5454885541e36c6e
+      - cargo-sources.json
+    build-commands:
+      - cargo --offline fetch --manifest-path Cargo.toml --verbose
+      - cargo build --offline --release --all-features
+      - install -D target/release/emojiclu /app/bin/emojiclu
+      - install -D target/release/resources/icons/hicolor/16x16/apps/io.github.timcharper.EmojiClu.png /app/share/icons/hicolor/16x16/apps/io.github.timcharper.EmojiClu.png
+      - install -D target/release/resources/icons/hicolor/24x24/apps/io.github.timcharper.EmojiClu.png /app/share/icons/hicolor/24x24/apps/io.github.timcharper.EmojiClu.png
+      - install -D target/release/resources/icons/hicolor/32x32/apps/io.github.timcharper.EmojiClu.png /app/share/icons/hicolor/32x32/apps/io.github.timcharper.EmojiClu.png
+      - install -D target/release/resources/icons/hicolor/48x48/apps/io.github.timcharper.EmojiClu.png /app/share/icons/hicolor/48x48/apps/io.github.timcharper.EmojiClu.png
+      - install -D target/release/resources/icons/hicolor/64x64/apps/io.github.timcharper.EmojiClu.png /app/share/icons/hicolor/64x64/apps/io.github.timcharper.EmojiClu.png
+      - install -D target/release/resources/icons/hicolor/128x128/apps/io.github.timcharper.EmojiClu.png /app/share/icons/hicolor/128x128/apps/io.github.timcharper.EmojiClu.png
+      - install -D target/release/resources/icons/hicolor/256x256/apps/io.github.timcharper.EmojiClu.png /app/share/icons/hicolor/256x256/apps/io.github.timcharper.EmojiClu.png
+      - install -D target/release/resources/icons/hicolor/512x512/apps/io.github.timcharper.EmojiClu.png /app/share/icons/hicolor/512x512/apps/io.github.timcharper.EmojiClu.png
+      - install -D packaging/flatpak/config/io.github.timcharper.EmojiClu.desktop /app/share/applications/io.github.timcharper.EmojiClu.desktop
+      - install -D packaging/flatpak/metainfo/io.github.timcharper.EmojiClu.appdata.xml /app/share/metainfo/io.github.timcharper.EmojiClu.appdata.xml
```

Comment:
how about having a buildsystem to do this work automatically?

Reply:
I don't understand, what would that look like?

Reply:
it would look like a build system. That when you ask it to install will install these in the right location, and in a way to is not specific to flatpak. (ie by setting a prefix)

the GNOME project has template for GNOME applications, including some that support Rust.

Reply:
if you repeat any of these step manually during the debian package creation that because you need a build system. One that will have an install rule and install file in their location supporting both `PREFIX` and `DESTDIR`. It's like a multi-decade old common practice. Nothing new.